weechat-android/src/main/java/com/ubergeek42/WeechatAndroid/WeechatActivity.java

```diff
@@ -102,11 +105,15 @@
 
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
-        if (DEBUG_LIFECYCLE) logger.debug("onCreate({})", savedInstanceState);
-        super.onCreate(savedInstanceState);
+        logger.debug("onCreate({})", savedInstanceState);
+
+        // after OOM kill and not going to restore anything? remove all fragments & open buffers
+        if (!P.isServiceAlive() && !BufferList.hasData()) {
```

Comment:
I feel like SharedPreferences is not the correct place to store service state. While it works, we've now got global state stored in Android's preference database, and have introduced global state which should instead be extracted from components via interfaces.